### PR TITLE
refactor: use passwords for auth

### PR DESCRIPTION
**Pull request by arrocke**
## What has changed

* Added passwords to accept invites and login routes and views
* Added length and confirmation validators to the text input
* Modified the error message component to be more flexible. When you use it, you have to specify each error message.
* Pulled out the users translation strings into a separate json file

## Connected Issues

relates to #124 

## QA Steps

1. Add the following value to `hashed_password` in your admin user in the database:
`s2:h2I35PXwsp81qZNX:fcb5e8e8bbb2b6bd93cad7d4af84c8057f58254d45574a3809fb3455ca88a7ee6d4690c89ea6c63e66bda1f5229da8b95f4227e5a6e21cac6acd83dcd4b730c7`
2. Try logging in with password `asdf1234`
3. Invite a user and confirm that the password you used to sign up works to log in

## Post-Deployment

We'll need to add passwords to existing users

### CODE REVIEW

File: packages/api/public/spec/paths/auth.yml [hunk 1]
```diff
@@ -1,18 +1,4 @@
...
-        in: query
-        description: the single use token
-      - name: redirectUrl
-        in: query
-        description: where to redirect to when login succeeds.
-    responses:
-      302:
-        description: Redirect
   post:
     summary: Login with magic link
```
Comment (Pertempto) [inline packages/api/public/spec/paths/auth.yml]: This needs updated

File: packages/web/src/public/locales/ar/translation.json [hunk 2]
```diff
@@ -1,47 +1,28 @@
 {
-  "language_one": "لغة",
+  "language_one": "اللغة",
```
Comment (Pertempto) [inline packages/web/src/public/locales/ar/translation.json:2]: Why so many changes in this file?

**Review by Pertempto — CHANGES_REQUESTED** [review]
I'm impressed with how few API changes were required to support passwords!  

# Improvements
* I think we could have a better login error message when users give an incorrect password.
* The browser doesn't recognize the email field on the accept invite page, because it isn't an input field. This breaks the built-in password manager because it tries to save the username as the email.  
    * I think maybe we could fix this by making the email a read-only input field? Maybe some additional annotations are required to make the browser recognize it.
    * I was testing this in Vivaldi, a chromium browser, so this might be specific to the chromium family tree of browsers.
    * EDIT: the same behavior occurs in Firefox.

# Bugs
* When a user accepts an invite and submits their username/password, we still get the incorrect flash.
- If a user has no password in the DB, they can login with an empty password. Even a brand-new invited user can do this before they use their invite link.

File: packages/web/src/public/locales/ar/translation.json (hunk 2 above)
Comment (arrocke) [reply packages/web/src/public/locales/ar/translation.json:2]: I didn't want to reorganize the file, so I just asked chatgpt to translate the entire file. It apparently came up with something different than google translate for a few of the keys

**Comment by arrocke** [comment]
@Pertempto Great feedback. I think I've addressed it all

File: packages/web/src/public/locales/ar/translation.json (hunk 2 above)
Comment (Pertempto) [reply packages/web/src/public/locales/ar/translation.json:2]: Ahh, interesting! I find it tedious to use Google Translate manually, so I usually generate translations with this extension: https://marketplace.visualstudio.com/items?itemName=funkyremi.vscode-google-translate

**Review by Pertempto — APPROVED** [review]
This is looking pretty good!

One small thing that needs finished before the merge:
- `invite_user` doesn't get translated correctly in `ManageLanguageView`

**Pull request merged**